src/main.ts

```diff
@@ -11,50 +10,34 @@ import { chmod } from 'fs/promises'
 export async function run(): Promise<void> {
   try {
     const pklVersion = core.getInput('pkl-version')
-    const githubToken = core.getInput('github-token')
+    core.debug(`Installing Pkl version ${pklVersion}`)
 
-    if (!githubToken) {
-      throw new Error('Input "github-token" is required')
-    }
+    // Try to retrieve from cache first
+    let cachedPath = tc.find('pkl', pklVersion)
+    if (cachedPath) {
+      core.debug(`Found cached Pkl version at ${cachedPath}`)
+    } else {
+      const assetName = findAssetName()
+      const downloadUrl = `https://github.com/apple/pkl/releases/download/${pklVersion}/${assetName}`
```

Comment:
This is fancy.
Nice one 👏 

But what if GitHub decided to change that URL?
Isn't it saver to use the release stuff with octokit?

Reply:
Understandable, but GitHub's release URLs have been stable for years, and it's highly unlikely they'll ever change. Using the direct URL keeps things simple and works well. IMO, using Octokit adds more friction without much benefit, given the stability of these URLs